controller_manager/src/controller_manager.cpp

```diff
@@ -440,6 +472,16 @@ bool ControllerManager::switchController(const std::vector<std::string>& start_c
     return false;
   }
 
+  if (!robot_hw_->canSwitch(start_list, stop_list))
+  {
+    ROS_ERROR("Could not switch controllers, due to mode conflicts");
```

Comment:
I see that you try to maintain parallelism with the resource conflict message, but I find the error message not so clear. Now there's no explicit notion of _modes_, or what a _mode conflict_ is. Here's some alternatives.

More explicit in blaming the robot for the failure:
```
"Could not switch controllers, requested controller state cannot be realized by robot"
```
Less explicit, more compact:
```
"Could not switch controllers, requested controller state is unfeasible"
```
Most verbose:
```
"Could not switch controllers. The resource combination for the requested controller state is unfeasible"
```